servers/parser-graphql-wrapper/src/readerView/readersSlug.ts

```diff
@@ -0,0 +1,32 @@
+import { itemIdFromSlug } from './idUtils';
+import DataLoader from 'dataloader';
+import { ItemLoaderType } from '../dataLoaders';
+
+export async function urlFromReaderSlug(
+  slug: string,
+  itemIdLoader: DataLoader<string, ItemLoaderType>,
+): Promise<string | null> {
+  const id = itemIdFromSlug(slug);
+  if (id == null) {
+    return null;
+  }
+  const itemLoaderResult = await itemIdLoader.load(id);
+
+  if (itemLoaderResult == null || itemLoaderResult.url == null) {
+    return null;
+  }
+
+  return itemLoaderResult.url;
+}
+
+/**
+ * Extract the e code from a URL. If the URL is a valid reader sharable link URL,
+ * return the encoded item id. Otherwise return undefined (e.g. if the URL is not a pocket reader URL).
+ * @param url the url to attempt to extract the encoded item id from
+ * @returns undefined if there is no slug to get or the encoded item id if it is a reader url
+ */
+export function extractSlugFromReadUrl(url: string): string | undefined {
+  const regex = /^(?:https?:\/\/)?getpocket\.com\/read\/((?!\d+$)[^/?#]+)/;
```

Comment:
can you explain why you included special characters here? the ID should not contain them

Reply:
I'm pretty sure this is what you want:

`/^(?:https?:\/\/)?getpocket\.com\/read\/([A-Za-z\d]+_[A-Za-z\d]+)$/gm`

Reply:
With test cases: 
![image](https://github.com/Pocket/pocket-monorepo/assets/34227334/72c0a568-6c60-4932-849b-567cded296bd)

Reply:
optionally allowing an ending `/` (not in capturing group)

`^(?:https?:\/\/)?getpocket\.com\/read\/([A-Za-z\d]+_[A-Za-z\d]+)\/?$`

Reply:
resolving -- we did more regexing in slack